### PR TITLE
Initial commit for ROS2 with readme info

### PR DESCRIPTION
**Pull request by aimarket**
Starting active development of the ros2 configuration.
The RaspberryPI `/src` folder will be removed if ROS2 ends up working properly

### CODE REVIEW

**Comment by aimarket** [comment]
Remove build files

**Comment by aimarket** [comment]
@nsk126 Hello,
I was able to find a `.gitingone` file that removed the build file along with many other unnecessary files, With my testing i was successfully able to 
- clone the repo
- build the Workspace with `colcon build` command
- and run the `ros2 run joy_controller joy_node` command to give me the desired output.

File: Readme.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# ROV
+
+## ros2_ws
+This is the ROS2 workspace for the ROV. It contains the following packages:
+- joy_controller - A package that will take in joystick input and publish it to the /joy topic
+- **TODO** - create basic file structure for the rest of the packages
+
```
Comment (MrPanda1) [inline Readme.md:34]: Add a `## Getting Started` section with some instructions (i.e. run colcon build in the ros2_ws folder. Downgrade setuptools if needed, how to test install is working, sourcing the `setup.sh` file, etc).

File: ros2_ws/install/setup.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# generated from colcon_core/shell/template/prefix_chain.sh.em
```
Comment (MrPanda1) [inline ros2_ws/install/setup.sh:1]: Add some instruction to README so people know to source this/add to `~/.bashrc`

File: ros2_ws/src/joy_controller/joy_controller/my_first_node.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
+import rclpy
+from rclpy.node import Node
+
+# Create a class that inherits from the Node class
+# The __init__ method is the constructor of the class
+class MyNode(Node):
```
Comment (MrPanda1) [inline ros2_ws/src/joy_controller/joy_controller/my_first_node.py]: This isn't a publisher or a subscriber. It just spins?

File: ros2_ws/src/joy_controller/test/test_pep257.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2015 Open Source Robotics Foundation, Inc.
```
Comment (MrPanda1) [inline ros2_ws/src/joy_controller/test/test_pep257.py]: We are adopting this coding standard?

File: ros2_ws/src/joy_controller/test/test_flake8.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
```
Comment (MrPanda1) [inline ros2_ws/src/joy_controller/test/test_flake8.py]: We are adopting this coding standard?

**Review by MrPanda1 — CHANGES_REQUESTED** [review]
### Use Virtual Envs
If you are going to predominantly be using Python here, I recommend setting up a Python virtual environment (see something like [PipEnv](https://pipenv.pypa.io/en/latest/)). This will help make sure everyone is running with the same packages and eliminate issues down the line.

### Linux only
Why are there `*.ps1`, `*.bash`, `*.zsh`, etc files if we are all using the same Linux box? Are these files needed?

**Comment by MrPanda1** [comment]
Instructions for those of you who may be confused about how to clone this down:
```
git checkout main
git pull origin pull/1/head
```

In general, you can do the following:
```
git checkout -b <name of branch on fork>
git pull origin pull/<PR #>/head
```

File: ros2_ws/src/joy_controller/joy_controller/my_first_node.py (hunk 3 above)
Comment (aimarket) [reply ros2_ws/src/joy_controller/joy_controller/my_first_node.py]: At the moment this is just placeholder code which will eventually be publishing to the jot topic with controller input.

File: ros2_ws/src/joy_controller/test/test_flake8.py (hunk 5 above)
Comment (aimarket) [reply ros2_ws/src/joy_controller/test/test_flake8.py]: file marked for deletion

File: ros2_ws/src/joy_controller/test/test_pep257.py (hunk 4 above)
Comment (aimarket) [reply ros2_ws/src/joy_controller/test/test_pep257.py]: file marked for deletion

File: ros2_ws/install/setup.sh (hunk 2 above)
Comment (aimarket) [reply ros2_ws/install/setup.sh:1]: task has been assigned in trello

File: Readme.md (hunk 1 above)
Comment (aimarket) [reply Readme.md:34]: assigned in trello

**Pull request merged**